### PR TITLE
dep(): `populateWithProperties` => `pick`

### PR DESCRIPTION
**Pull request by ShaMan123**
signature changed

`pick(source, keys) => new object`

### CODE REVIEW

**Review by ShaMan123 — COMMENTED** [review]
READY

**Comment by asturur** [comment]
no leave the util folder aloneeeee i m working on that

**Comment by ShaMan123** [comment]
OK
but please make this function so that it returns a new object and doesn't mutate anything!

**Comment by asturur** [comment]
the pr is good do it when i l done there. I have 2 files left

**Comment by asturur** [comment]
> OK but please make this function so that it returns a new object and doesn't mutate anything!

The old populateWithProperties was mutating objects that were created to be returned.
It wasn't bad per se. I agree that pick may be more useful and reusable than populateWithProperties that was a pick + assign.

**Comment by asturur** [comment]
The typing for pick is a bit too strict, as it is toObject.
We don't need to revert it now, but you have to consider a common thing that happen is that devs may want to add id, name or other values that are not part of the class, and export it.

File: src/static_canvas.class.ts [hunk 1]
```diff
@@ -1027,30 +1028,19 @@ import { removeFromArray } from './util/internals';
...
+      const clipPathData = clipPath && !clipPath.excludeFromExport ?
+        this._toObject(clipPath, methodName, propertiesToInclude) :
+        null;
+      return {
         version: VERSION,
-        objects: this._toObjects(methodName, propertiesToInclude),
+        ...pick(this, propertiesToInclude),
+        objects: this._objects
+          .filter((object) => !object.excludeFromExport)
+          .map((instance) => this._toObject(instance, methodName, propertiesToInclude)),
```
Comment (asturur) [inline src/static_canvas.class.ts:1040]: not sure this is an improvement whatsover. is a bit harsh to read

**Review by asturur — APPROVED** [review]
I ll open a task to revert the strict typing.

**Comment by ShaMan123** [comment]
I don't understand your comment regarding the typing of pick.
What is strict about it?

**Pull request merged**